pkg/controller/cainjector/setup.go

```diff
@@ -58,12 +60,41 @@ var (
 		listType:     &apiext.CustomResourceDefinitionList{},
 	}
 
-	injectorSetups  = []injectorSetup{MutatingWebhookSetup, ValidatingWebhookSetup, APIServiceSetup, CRDSetup}
+	AuditSinkSetup = injectorSetup{
+		resourceName: "auditsink",
+		injector:     auditSinkTarget{},
+		listType:     &v1alpha1.AuditSinkList{},
+	}
+
+	injectorSetups  = []injectorSetup{MutatingWebhookSetup, ValidatingWebhookSetup, APIServiceSetup, CRDSetup, AuditSinkSetup}
 	ControllerNames []string
 )
 
-// Register registers an injection controller with the given manager, and adds relevant indices.
-func Register(mgr ctrl.Manager, setup injectorSetup, sources ...caDataSource) error {
+// registerAllInjectors registers all injectors and based on the
+// graduation state of the injector decides how to log no kind/resource match errors
+func registerAllInjectors(mgr ctrl.Manager, sources ...caDataSource) error {
+	for _, setup := range injectorSetups {
+		if err := registerInjector(mgr, setup, sources...); err != nil {
+			if meta.IsNoMatchError(err) {
+				if setup.injector.IsAlpha() {
+					ctrl.Log.Info("unable to register injector which is still in an alpha phase."+
+						" Enable the feature on the API server in order to use this injector",
+						"injector", setup)
```

Comment:
When looking through the log messages from the `cainjector`, I see: `2020-03-18T11:01:29.819491669Z stderr F I0318 11:01:29.819346       1 setup.go:80] cert-manager "msg"="unable to register injector which is still in an alpha phase. Enable the feature on the API server in order to use this injector"  "injector"={}` - I think passing `setup` here is not sufficient to print useful information to the user. This message is a bit misleading and could lead users to think some genuine errors has happened...

Can you instead use `setup.resourceName` here?